### ISSUE TITLE
After updating to EXPO 44, Linking.createURL does not return the expected string  when using an iOS simulator

### ISSUE THREAD

**Issue by msevestre** [expo/expo#16834]
### Summary

After upgrading to expo 44, I realize  that the method `Linking.makeURL` was deprecated. `Linking.createURL` should be used instead.

However the behavior of those two methods is not the same (at least in dev mode)

```
  const redirectUrl = Linking.makeUrl('callback');
  console.log({ redirectUrl });

 Object {
  "redirectUrl": "exp://10.0.0.111:19000/callback",
}
```

```
  const redirectUrl = Linking.createUrl('callback');
  console.log({ redirectUrl });

 Object {
  "redirectUrl": "exp://10.0.0.111:19000callback",
}
```

as you can see, the `/` between scheme and path is missing. Is this the expected behavior?
 

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

44

### Environment

expo-env-info 1.0.2 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.18.2 - /usr/local/bin/node
      Yarn: 1.22.0 - /usr/local/bin/yarn
      npm: 6.14.5 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6241897
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.6 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      eas-cli: 0.48.2
      expo-cli: 5.3.0
    Expo Workflow: managed

### Reproducible demo

I do not have a demo that I can provide at the moment. I want to know if this behavior is as expected

**Comment by msevestre** [comment]
Stale but this is still happening with Expo 45

**Comment by msevestre** [comment]
Problem still persists lol.

**Comment by stefanschweng** [comment]
I had a similar issue. When I was using Expo SDK 45, `Linking.createURL` was working as expected until I migrated from Classic Updates (i.e. `expo publish`) to EAS Updates (https://docs.expo.dev/eas-update/migrate-to-eas-update/). I figured that setting the config property `expo.updates.url` (by running `eas update:configure`) "destroyed" something and `Linking.createURL` gave me the same results as you reported it.
I just upgraded to Expo SDK 46 and everything works as expected again.
